eventual.go

```diff
@@ -56,11 +66,23 @@ func (v *value) Set(i interface{}) {
 		for _, waiter := range v.waiters {
 			waiter <- i
 		}
+		v.waiters = make([]chan interface{}, 0)
 		v.set = true
 	}
 	v.m.Unlock()
 }
 
+func (v *value) Reset() {
+	v.m.Lock()
+	v.v = nil
+	v.set = false
+	for _, waiter := range v.waiters {
+		close(waiter)
+	}
+	v.waiters = make([]chan interface{}, 0)
```

Comment:
Why clear the waiters out?  What's wrong with them continuing to wait for the value to be set?  I may be missing the point of the `Reset` method.

Reply:
To expand on my feeling of missing the point: I guess to me an "eventually available" value shouldn't at some point in time transition from available to unavailable.  I think I'm kind of getting hung up on that and unable to understand where this would be useful, but I'm sure you've got a use case.

Reply:
Yeah, that's a good point. Will change.